compiler/src/iree/compiler/Dialect/Flow/Transforms/FusionUtils.cpp

```diff
@@ -27,9 +33,15 @@ namespace Flow {
 static bool canInsOperandTieWithOutsOperand(OpOperand *insOperand) {
   auto linalgOp = dyn_cast<linalg::LinalgOp>(insOperand->getOwner());
   if (!linalgOp) return false;
+
   AffineMap insOperandIndexingMap = linalgOp.getTiedIndexingMap(insOperand);
+
   auto canTieWithOutsOperand = [&](OpOperand *outsOperand) {
-    if (linalgOp.getTiedIndexingMap(outsOperand) != insOperandIndexingMap) {
+    auto outputMap = linalgOp.getTiedIndexingMap(outsOperand);
+    if (clFuseTranspose) {
```

Comment:
I am confused about this change. We should not need to do this. With the intechange pass, the dispatch region formation should fuse the ops directly (that is why we did the intechange). You need one or the other but not both. Can you give me an example where after interchange things still didnt fuse?

Reply:
The transpose would have the following indexing maps:

input: (d0, d1) -> (d0, d1)
output: (d0, d1) -> (d1, d0)

Thus, the linalg op's input indexing map is different from the output indexing map, and the fusion is not triggered with the original code.

Reply:
Ah I see. Thanks! I'd actually hoist this option usage out of this method (this method is a hack which we might not need anymore, need to evaluate). So move this to line 79 and rename the flag

1) `iree-flow-fuse-transpose` -> `iree-flow-ensure-inplacable-consumer` . Make the new flag default true.
2) At line 79
```
return !clEnsureInplacable || canInsOperandTieWithOutsOperand
```

Reply:
Everything in this method (`canInsOperandTieWithOutsOperand`) is a WAR that wont work in the general case. So I am saying we add a flag to side-step this entire method for exploration of fusion with transpose-like ops (which is enabled by this change)